### ISSUE TITLE
The method upload() can return the promise object ?

### ISSUE THREAD

**Issue by ruanhaojian** [nervgh/angular-file-upload#620]
or how can i  get the promise obj?

**Comment by nervgh** [comment]
You could write something like this:
```js
item.uploadPromise = function() {
    let deferred = $q.defer(); // you need inject the $q
    this.onSuccess = deferred.resolve;
    this.onError = deferred.reject;
    this.upload();
    return deferred.promise;
}
```
See also https://github.com/nervgh/angular-file-upload/wiki/FAQ#5-how-i-can-modify-native-methods-or-add-new-features-to-fileuploader

**Comment by prindacerk** [comment]
Could you please explain this a bit better for me? This seems to be for item specific right? Under which method should this be implemented?

Situation is this. I have server side processing happening for the file that gets uploaded. Now your progress shows 100% since the file has been sent to the server. But the process is not completed on server end. It waits on 100% for the server response. Users meanwhile can click on cancel/remove and they won't realize it has been processed on server end.

I want to show busy (using angular-busy module) so the users won't click on anything once file has been sent to the server. But that module works by tying itself to promise. So I need to get the promise object. It could be tied to the http request also but since that is not exposed in the fileuploader instance, I need another way.

Can you please give an example of how this is supposed to work? Or explain what I should do to enable the busy loading until server response has returned?

**Comment by mrutz** [comment]
This is a pretty critical feature for a file upload module to have. Can we please get a response on this?
